### ISSUE TITLE
GraphQL::Type::List needs method "perl_to_graphql"

### ISSUE THREAD

**Issue by schroeer** [graphql-perl/graphql-perl#28]
I believe that `GraphQL::Type::List` needs a methods `perl_to_graphql` for when a [`default_value`](https://github.com/graphql-perl/graphql-perl/blob/213103dd9c232d567c68ac5a396ac9c0b71b6ba8/lib/GraphQL/Type/Library.pm#L96) is provided for a list-type argument.

**Comment by mohawk2** [comment]
As shown in the commit linked above, this currently appears to work right. Is this a theoretical concern, or do you have a test case that I can use to fix an actual problem?

**Comment by schroeer** [comment]
Sorry. I should have provided more information in the first place.

The test program below fails. Probably related, GraphiQL cannot show the schema for my `Mojolicious::Plugin::GraphQL` application when the `default_value` is present.

```
#!perl

use strict;
use GraphQL::Schema;
use GraphQL::Type::Scalar qw($String);
use GraphQL::Type::Object;

my $q = GraphQL::Type::Object->new(
    name => 'Query',
    fields => {
        a  => {
            type => $String,
            args => {
                b => {
                    type => $String->list,
                    default_value => [qw/foo bar/],
                },
            },
            resolve => sub {
                return "foobar";
            },
        },
    },
);

my $schema = GraphQL::Schema->new(query => $q);

print $schema->to_doc();
```

**Comment by mohawk2** [comment]
Providing a repro case has to be considered "the basics" :-D To split a hair, I now think what you were correctly saying is that it's needed not when providing it, but when trying to retrieve it.

I also fixed this for InputObject. Released as 0.42. Thanks for the report!
